foreshadow/cleaners/data_cleaner.py

```diff
@@ -79,22 +101,44 @@ def pick_transformer(self, X, y=None, **fit_params):
         best_score = 0
         best_cleaner = None
         for cleaner, name in cleaners:
-            cleaner = cleaner(column_sharer=self.column_sharer, name=name)
+            cleaner = cleaner(column_sharer=self.column_sharer)
             score = cleaner.metric_score(X)
             if score > best_score:
                 best_score = score
                 best_cleaner = cleaner
-
         if best_cleaner is None:
             return NoTransform(column_sharer=self.column_sharer)
         return best_cleaner
 
+    def resolve(self, X, *args, **kwargs):
+        """Resolve the underlying concrete transformer.
+
+        Sets self.column_sharer with the domain tag.
+
+        Args:
+            X: input DataFrame
+            *args: args to super
+            **kwargs: kwargs to super
+
+        Returns:
+            Return from super.
+
+        """
+        ret = super().resolve(X, *args, **kwargs)
+        if self.column_sharer is not None:
+            self.column_sharer[
+                "domain", X.columns[0]
+            ] = self.transformer.__class__.__name__
```

Comment:
Minor comment: should we define a more readable domain field in the concrete transformer class? `DollarFinancialCleaner` may be an acceptable domain name, but `DropCleaner` domain seems a bit hard to interpret.

Reply:
Yes, I see that being the correct approach as well.
My thought process was to design this in hand with FeatureEngineerer when the logic there get's implemented, as until there we have no steps pulling from the domain key yet. Would you like me to do that now?

Reply:
issue created